app/models/miq_provision_virt_workflow.rb

```diff
@@ -540,16 +539,10 @@ def allowed_templates(options = {})
       unless tag_conditions.blank?
         _log.info "Filtering VM templates with the following tag_filters: <#{tag_conditions.inspect}>"
         vms = MiqTemplate.find_tags_by_grouping(tag_conditions, :ns => "/managed", :conditions => condition)
-        if vms.blank?
-          _log.warn "tag_filters returned an empty VM template list.  Tag Filters used: <#{tag_conditions.inspect}>"
-          run_search = false
-        else
-          vms.each { |vm| _log.info "tag_filter template returned: <#{vm.id}:#{vm.name}>  GUID: <#{vm.guid}>  UID_EMS: <#{vm.uid_ems}>" }
```

Comment:
this calls a `count(*)` and a `to_a` to fetch these records.
But this is before we've applied our conditions. so the work si for no reason.

If you want this logging in here, I can put into a block that check `_log.info?` first.